File: Python/953.verifying_an_alien_dictionary.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        english = "abcdefghijklmnopqrstuvwxyz"
        table = dict(zip(order, english))
        for i in range(len(words) - 1):
            word1, word2 = "", ""
            for char in words[i]:
                word1 += table[char]
            for char in words[i+1]:
                word2 += table[char]
            if word1 > word2:
                return False
        return True

File: C++/Hash Table/953. Verifying an Alien Dictionary.cpp
class Solution {
public:
    
    bool isAlienSorted(vector<string>& words, string order) {
        vector<int> alphabet(26);
        for (int i = 0; i < size(order); i++) {
            alphabet[order[i] - 'a'] = i;
        }
        for (int i = 0; i < size(words) - 1; i++) {
            string w1 = words[i], w2 = words[i+1];
            if (w1.find(w2) != -1 && size(w1) > size(w2)) return false;
            int j = 0, k = 0;
            bool legit = true;
            while(j < size(w1) && k < size(w2)) {
                if (w1[j] != w2[k]) {
                    legit = alphabet[w1[j] - 'a'] < alphabet[w2[k] - 'a'];
                    break;
                }
                else {
                    j++;
                    k++;
                }
            }
            if (!legit) {
                return false;
            }
        }
        return true;
    }
};